### PR TITLE
[Documentation] Update examples for GlowingOverscrollIndicator

### LINKED ISSUE #76489: GlowingOverscrollIndicator dartpad samples are confusing
GlowingOverscrollIndicator code samples may be confusing when ran in web, because glow is not drawn

https://api.flutter.dev/flutter/widgets/GlowingOverscrollIndicator-class.html

TagertPlatform should be used in these samples

### PR DESCRIPTION
**Pull request by TahaTesser**
Updated existing samples and add a new example to demonstrate how to achieve GlowingOverscrollIndicator on devices with mouse and touch device pointer such as Desktop

Fixes https://github.com/flutter/flutter/issues/76489

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 1]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
+/// class MyCustomScrollBehavior extends MaterialScrollBehavior {
+///   @override
+///   Set<PointerDeviceKind> get dragDevices => {
+///     PointerDeviceKind.touch,
+///     PointerDeviceKind.mouse,
+///   };
+/// }
+///
+/// void main() => runApp(const MyApp());
+/// // This is the main application widget.
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:71]: This comment isn't needed

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 2]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
+///           // to Android to get Android scroll behavior.
+///           platform: TargetPlatform.android,
+///       ),
+///       home: const Scaffold(
+///         body: MyStatelessWidget(),
+///       ),
+///     );
+///   }
+/// }
+/// // This is the stateless widget that the main application instantiates.
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:93]: This comment isn't needed

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 3]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
+///
+///   static const String _title = 'Flutter Code Sample';
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       title: _title,
+///       scrollBehavior: MyCustomScrollBehavior(),
+///       theme: ThemeData(
+///           // Set [TargetPlatform](https://api.flutter.dev/flutter/foundation/TargetPlatform-class.html)
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:83]: two space indent

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 4]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
+///           platform: TargetPlatform.android,
+///       ),
+///       home: const Scaffold(
+///         body: MyStatelessWidget(),
+///       ),
+///     );
+///   }
+/// }
+/// // This is the stateless widget that the main application instantiates.
+/// class MyStatelessWidget extends StatelessWidget {
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:94]: It would be helpful to give this class a name that indicates what it's for.

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 5]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
+///     PointerDeviceKind.mouse,
+///   };
+/// }
+///
+/// void main() => runApp(const MyApp());
+/// // This is the main application widget.
+/// class MyApp extends StatelessWidget {
+///   const MyApp({Key? key}) : super(key: key);
+///
+///   static const String _title = 'Flutter Code Sample';
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:75]: The title and this constant can be left out, since they don't really add to the sample.

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 6]
```diff
@@ -46,69 +46,154 @@ import 'ticker_provider.dart';
...
-///         const SliverAppBar(title: Text('Custom PaintOffset')),
-///         SliverToBoxAdapter(
-///           child: Container(
-///             color: Colors.amberAccent,
-///             height: 100,
-///             child: const Center(child: Text('Glow all day!')),
+/// import 'package:flutter/gestures.dart';
+/// import 'package:flutter/material.dart';
+///
+/// // Override the [default drag behavior](https://flutter.dev/docs/release/breaking-changes/default-scroll-behavior-drag)
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:60]: This comment looks like it belongs to a separate sample that _just_ shows how to override the default scroll behavior.

**Review by HansMuller — COMMENTED** [review]
It looks like maybe you meant to introduce a second sample  that _just_ shows how to override the default scroll behavior. That would clearer than mixing the two samples together.

**Comment by TahaTesser** [comment]
@HansMuller 
Update the PR with the changes and add new sample to demo overriding default behavior

**Comment by goderbauer** [comment]
@TahaTesser Looks like there's a problem with the docs, could you please fix that?

```
dartdoc:stdout: Generating docs for library widgets from package:flutter/widgets.dart...
dartdoc:stderr:   error: unresolved doc reference [default drag behavior]
dartdoc:stderr:     from widgets.GlowingOverscrollIndicator: (file:///tmp/flutter%20sdk/packages/flutter/lib/src/widgets/overscroll_indicator.dart:234:7)
```

**Comment by TahaTesser** [comment]
> @TahaTesser Looks like there's a problem with the docs, could you please fix that?
> 
> ```
> dartdoc:stdout: Generating docs for library widgets from package:flutter/widgets.dart...
> dartdoc:stderr:   error: unresolved doc reference [default drag behavior]
> dartdoc:stderr:     from widgets.GlowingOverscrollIndicator: (file:///tmp/flutter%20sdk/packages/flutter/lib/src/widgets/overscroll_indicator.dart:234:7)
> ```

Thanks, fixed it

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 12.
View them at https://flutter-gold.skia.org/cl/github/87078

**Comment by TahaTesser** [comment]
@HansMuller 
Added an example in api/example

File: examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      scrollBehavior: MyCustomScrollBehavior(),
+      theme: ThemeData(
+        // Set [TargetPlatform](https://api.flutter.dev/flutter/foundation/TargetPlatform-class.html)
```
Comment (HansMuller) [inline examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart:31]: If this is necessary then we should explain why. Otherwise, it doesn't seem relevant to the example.

File: examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart [hunk 8]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for GlowingOverscrollIndicator
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+// Override the [default drag behavior](https://flutter.dev/docs/release/breaking-changes/default-scroll-behavior-drag).
```
Comment (HansMuller) [inline examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart:10]: This isn't a helpful comment, since it just reiterates what's obvious. It would be helpful to note that we're adding support for PointerDeviceKind.mouse and it would be a little more correct to add that to super.dragDevices.

The API doc for dragDeviceProperties include a rather dire warning:

"Enabling this for PointerDeviceKind.mouse will make it difficult or impossible to select text in scrollable containers and is not recommended."

Is this example still advisable given the warning?

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 9]
```diff
@@ -66,6 +66,14 @@ import 'transitions.dart';
 /// ** See code in examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.1.dart **
 /// {@end-tool}
 ///
+/// {@tool dartpad}
+/// This example demonstrates how to override the default drag behavior to render
+/// `GlowingOverscrollIndicator` affect on devices with `PointerDeviceKind.touch`
+/// and `PointerDeviceKind.mouse`.
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:72]: MaterialScrollBehavior supports PointerDeviceKind.touch by default

**Review by HansMuller — COMMENTED** [review]
I'm not sure this is a good example to promote.

**Comment by TahaTesser** [comment]
Thanks for the feedback, I will close this for now.
I might come to this subject this in the future.

File: examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart (hunk 8 above)
Comment (TahaTesser) [reply examples/api/lib/widgets/overscroll_indicator/glowing_overscroll_indicator.2.dart:10]: That makes sense, noted.
